orc8r/cloud/go/services/analytics/collector_servicer.go

```diff
@@ -18,32 +18,127 @@ import (
+						if gatewayLabelPresent {
+							if numUsers, ok := userNetworkTable.UsersGatewayTable[gatewayID]; ok {
+								if numUsers > analyticsConfig.MinUserThreshold {
+									filteredResults = append(filteredResults, result)
+								} else {
+									glog.V(1).Infof("metric %s(gateway label) dropped,active users %d below user threshold %d",
+										result.GetMetricName(), numUsers, analyticsConfig.MinUserThreshold)
+								}
+							}
+						} else {
+							if userNetworkTable.TotalUsersPerNetwork > analyticsConfig.MinUserThreshold {
+								filteredResults = append(filteredResults, result)
+							} else {
+								glog.V(1).Infof("metric %s(network label) dropped,active users %d below user threshold %d",
+									result.GetMetricName(), userNetworkTable.TotalUsersPerNetwork, analyticsConfig.MinUserThreshold)
+							}
+						}
+					} else {
+						if userStateMgr.TotalUsers > analyticsConfig.MinUserThreshold {
+							filteredResults = append(filteredResults, result)
+						} else {
+							glog.V(1).Infof("metric %s(deployment) dropped, active users %d below user threshold %d", result.GetMetricName(),
+								userStateMgr.TotalUsers, analyticsConfig.MinUserThreshold)
+						}
+					}
+				} else {
+					filteredResults = append(filteredResults, result)
+				}
+			} else {
+				glog.V(1).Infof("Metric %s export configuration disabled", result.GetMetricName())
+			}
+		} else {
+			glog.Errorf("Metric Configuration not found for %s", result.GetMetricName())
+		}
+	}
+	return filteredResults
+}
+
+// registerResults registers the computed metric with Prometheus
+func (svc *CollectorService) registerResults(calcParams *calculations.CalculationParams, results []*protos.CalculationResult) {
+	filteredResults := []*protos.CalculationResult{}
+	analyticsConfig := svc.analyticsConfig
+	for _, result := range results {
+		if metricConfig, ok := analyticsConfig.Metrics[result.GetMetricName()]; ok {
+			if metricConfig.Register {
+				filteredResults = append(filteredResults, result)
+			}
+		} else {
+			glog.Errorf("Metric Configuration not found for %s", result.GetMetricName())
+		}
+	}
+	if len(filteredResults) > 0 {
+		glog.V(1).Info("Registering ", filteredResults)
+		calculations.RegisterResults(calcParams, filteredResults)
+	}
 }
 
 //Collect does the operation of running through calculations and returning results
 func (svc *CollectorService) Collect(context.Context, *protos.CollectRequest) (*protos.CollectResponse, error) {
+	// update the user state
+	if svc.userStateMgr != nil {
+		svc.userStateMgr.Update()
+	}
+
+	// collect the calculations
 	response := &protos.CollectResponse{}
 	for _, calc := range svc.calculations {
 		results, err := calc.Calculate(svc.promAPIClient)
 		if err != nil {
 			glog.Errorf("Error calculating metric: %s", err)
 			continue
 		}
-		glog.V(10).Infof("results are %v", results)
-		response.Results = append(response.Results, results...)
+
+		// check and register results
```

Comment:
Here and elsewhere: these sorts of "paragraph intro sentence" comments are a bad code smell. The code should stand on its own. If the encasing function becomes too long to understand just with proper naming and spacing, then break it into appropriate sub-functions with descriptive names